spring-boot-project/spring-boot/src/main/java/org/springframework/boot/logging/structured/StructuredLoggingJsonPropertiesBeanFactoryInitializationAotProcessor.java

```diff
@@ -58,10 +57,8 @@ public BeanFactoryInitializationAotContribution processAheadOfTime(ConfigurableL
 	}
 
 	private static String getCustomStackTracePrinter(StructuredLoggingJsonProperties properties) {
-		return Optional.ofNullable(properties.stackTrace())
-			.filter(StackTrace::hasCustomPrinter)
-			.map(StackTrace::printer)
-			.orElse(null);
+		StackTrace stackTrace = properties.stackTrace();
+		return (stackTrace != null && stackTrace.hasCustomPrinter()) ? stackTrace.printer() : null;
```

Comment:
The previous one sounded more readable to me i am curious to see what the spring team will think. But honestly at this point just use Moderne to replace every Optional of the project…

Reply:
> I am curious to see what the spring team will think.

Johnny already provided a reference, please see the link above.

Reply:
@snicoll I said that because when I git blamed it showed me you were the author of the code so it was two different rules by the Spring core developers 😄

Reply:
@snicoll my bad I have misread the git history I thought you were the author so I was confused about two different rules among the Spring team ☺️